CONTRIBUTING.md

```diff
@@ -1,189 +1,272 @@
+  Wait until the `API server listening at: [::]:2345` message appears
+  Start the "Debug vcluster (localhost:2346)" configuration in VSCode to connect your debugger session.
+  Note: vcluster won't start until you connect with the debugger.
+  Note: vcluster will be stopped once you detach your debugger session.
+
+TIP: hit an up arrow on your keyboard to find the commands mentioned above :)
+
+vcluster-0:vcluster-dev$ 
+```
+
+Now, your terminal is running in DevSpace, and you can develop and test with DevSpace. 
 
-#########################################################
-[info]   DevSpace UI available at: http://localhost:8090
-#########################################################
+### Developing and Testing vCluster using DevSpace 
 
-[0:sync] Waiting for pods...
-[0:sync] Starting sync...
-[0:sync] Sync started on /Users/fabiankramm/Programmieren/go-workspace/src/github.com/loft-sh/vcluster <-> . (Pod: vcluster/vcluster-0)
-[0:sync] Waiting for initial sync to complete
-[info]   Opening shell to pod:container vcluster-0:syncer
-root@vcluster-0:/vcluster#
+Start vcluster in DevSpace
+
+```
+vcluster-0:vcluster-dev$ go run -mod vendor cmd/vcluster/main.go start
 ```
 
-Then you can start vcluster with
+Now, you can start to work with the virtual cluster based on the source code. This vCluster is running on your local Kubernetes cluster.
+
+
+If you change a file locally, DevSpace will automatically sync the file into the Devspace container. After any changes, re-run the same command in the DevSpace terminal to apply the changes.
 
 ```
-go run -mod vendor cmd/vcluster/main.go start
+vcluster-dev$ go run -mod vendor cmd/vcluster/main.go start
 ```
 
-Now if you change a file locally, DevSpace will automatically sync the file into the container and you just have to rerun `go run -mod vendor cmd/vcluster/main.go start` within the terminal to apply the changes.
 
-You can access the vcluster by running `devspace enter` in a separate terminal:
+### Developing the hostpath-mapper Component with DevSpace
 
+In case you need to develop the hostpath-mapper daemonset instead of the syncer, you can use the `dev-hostpath-mapper` profile in `devspace.yaml`. 
+
+```
+$ devspace dev -p dev-hostpath-mapper
 ```
-devspace enter -n vcluster
 
-? Which pod do you want to open the terminal for? vcluster-0:syncer
-[info]   Opening shell to pod:container vcluster-0:syncer
-root@vcluster-0:/vcluster# kubectl get ns
-NAME              STATUS   AGE
-default           Active   2m18s
-kube-system       Active   2m18s
-kube-public       Active   2m18s
-kube-node-lease   Active   2m18s
-root@vcluster-0:/vcluster#
+This deploys the hostpath-mapper as a deployment instead of a daemonset and take care of other modifications to be made and allow you to develop the hostpath-mapper itself against the syncer.
+
+### Debug vCluster using Devspace and with Delve
```

Comment:
We might want to move/merge this with ### Developing and Testing vCluster using DevSpace above into a sub section? Similar to my suggestion above.